hackathon_site/hardware/views.py

```diff
@@ -31,9 +36,31 @@ def get(self, request, *args, **kwargs):
         return self.list(request, *args, **kwargs)
 
 
-class OrderListView(mixins.ListModelMixin, generics.GenericAPIView):
+class OrderListView(generics.ListCreateAPIView):
     queryset = Order.objects.all()
-    serializer_class = OrderSerializer
+    serializer_class = OrderListSerializer
+    serializer_method_classes = {
+        "GET": OrderListSerializer,
+        "POST": OrderCreateSerializer,
+    }
+
+    def get_serializer_class(self):
+        try:
+            return self.serializer_method_classes[self.request.method]
+        except (KeyError, AttributeError):
+            return super().get_serializer_class()
 
     def get(self, request, *args, **kwargs):
         return self.list(request, *args, **kwargs)
+
+    @transaction.atomic
+    @swagger_auto_schema(responses={201: OrderCreateResponseSerializer})
+    def post(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        create_response = serializer.save()
+        response_serializer = OrderCreateResponseSerializer(data=create_response)
+        response_serializer.is_valid()
+        response_data = response_serializer.data
+        headers = self.get_success_headers(response_data)
```

Comment:
I don't think this is actually going to do anything, since the `OrderCreateResponseSerializer` doesn't have a URL field (unless it magically derives one from order_id?)

Reply:
i didn't know what this does i just copied it from the generic implementation lool